editor/plugins/skeleton_3d_editor_plugin.cpp

```diff
@@ -922,7 +922,7 @@ void vertex() {
 
 	VERTEX = VERTEX;
 	POSITION = PROJECTION_MATRIX * VIEW_MATRIX * MODEL_MATRIX * vec4(VERTEX.xyz, 1.0);
-	POSITION.z = mix(POSITION.z, 0.0, 0.999);
+	POSITION.z = mix(POSITION.z, POSITION.w, 0.999);
```

Comment:
```suggestion
	POSITION.z = mix(POSITION.z, 1.0, 0.999);
```

Reply:
I believe it is correct to mix towards POSITION.w. The GPU will perform a perspective divide internally, effectively POSITION.xyz /= POSITION.w
The goal is to produce a value close to 1.0 post-perspective divide, so we want the GPU to calculate POSITION.z = 1.0 * POSITION.w / POSITION.w

Reply:
That's right

Reply:
If apply it, the order seems to be reversed.

before:
![image](https://github.com/godotengine/godot/assets/61938263/df45594b-11cb-47c2-8fc7-e92446f1785f)

apply it:
![image](https://github.com/godotengine/godot/assets/61938263/d2998a1e-f8d2-44b8-a84e-974e016c805b)